megatron/tokenizer/tokenizer.py

```diff
@@ -298,17 +314,23 @@ def vocab(self):
     def inv_vocab(self):
         return self.tokenizer._tokenizer.decoder
 
-    def tokenize(self, texts: Union[str, List[str]], context_length=2048):
+    def tokenize(self, texts: Union[str, List[str]]):
         if isinstance(texts, str):
             texts = [texts]
         texts = [whitespace_clean(basic_clean(text)) for text in texts]
         input_ids = self.tokenizer(
             texts,
-            return_tensors='pt',
-            max_length=context_length,
-            padding='max_length',
+            max_length=self.seq_length,
             truncation=True,
         ).input_ids
+        # add eod_id and pad with pad_id
```

Comment:
Sure, we can add the flag for add_eod. 
To ensure add_eod is meanful, could u provide a example that input of model don't have these eod id ?

And add_pad seems necessary, if we don't add it, the model can't run as expected.

Reply:
Will have to look for it. Its useful to keep it general though, since the tokenizing pipeline in GPT NeoX also provides an option not to add eod. 
add_pad by default sounds okay to me. The only issue with that I can see is that, one could technically just concatenate all teh seuences and cut them into sequence length parts like it is done currently? Could be more efficient use of resources?

Reply:
I am confused by "concatenate all the sequences and cut them into sequence length parts". 
Could u pls describe detailly about what does this mean and how it be more efficient?